upgrade.go

```diff
@@ -533,14 +569,20 @@ func (pth *programTestHelper) parseStackName(state string) string {
 
 func (b *providerUpgradeBuilder) providerUpgradeRecordBaselines(t *testing.T) {
 	info := b.newProviderUpgradeInfo(t)
-	ambientProvider, _ := exec.LookPath(b.providerBinary())
-	require.Emptyf(t, ambientProvider, "please remove the provider from PATH")
 	ensureFolderExists(t, info.recordingDir)
 	deleteFileIfExists(t, info.stateFile)
 	deleteFileIfExists(t, info.grpcFile)
+
 	test := integration.ProgramTestOptions{
 		Dir: b.program,
-		Env: append(os.Environ(), fmt.Sprintf("PULUMI_DEBUG_GRPC=%s", info.grpcFile)),
+		Env: append(os.Environ(),
+			// Make sure that local provider builds in PATH do not interfere with
+			// recording baseline versions.
+			"PULUMI_IGNORE_AMBIENT_PLUGINS=true",
```

Comment:
We set this here, and we have also added a way to carefully configure ambient plugins. Can you explain why these don't interfere with each other?

Reply:
That's a good catch. They actually do interfere. We need some form of testing here, ugh, probably capturing ProgramTest invocations (and cmd.Exec) invocations.

IN the case of Node, we want PULUMI_IGNORE_AMBIENT_PLUGINS=true since we'll be installing baseline versions through YARN.

IN the case of YAML though, no such luck, YAML should not set  PULUMI_IGNORE_AMBIENT_PLUGINS as it relies on them to be in PATH.

Reply:
I've been testing the YAML upgrade things manually but it's really hard to spot which version ends up getting used, and it's really important...